src/App.mc

```diff
@@ -21,7 +23,9 @@ function loadProviders() {
       } catch (exception) {
         var msg = exception.getErrorMessage();
         log(ERROR, msg);
-        displayError(msg);
+        if (displayerror) {
+          displayError(msg);
+        }
```

Comment:
We can drop the error "rendering" of this class of errors completely IMO. There is this error reporting feature of Garmin (ERA?) which reports uncaught exceptions.. unless we cannot trigger it explicitly, we could consider letting the app crash once here, i.e.
* catch errors
* drop the faulty entries from Application.Storage
* rethrow when there were entries dropped

Or do you have a better idea in how to handle this?/

Reply:
Please also see my comment below on why we need to avoid the call to `displayError()` here.

Reply:
@shinji What was the reason why you needed to not call `displayError` in the glance view? As I do not have a device with glance support I can't reproduce exactly what would go wrong.

Reply:
@ch1bo GlanceView has a smaller canvas, and run with limited memory allocated, e.g. 32KB for most devices.
I will try to display error in GlanceView and compare memory usage...
https://forums.garmin.com/developer/connect-iq/b/news-announcements/posts/widget-glances---a-new-way-to-present-your-data
https://developer.garmin.com/connect-iq/core-topics/glances/

Reply:
No need.. I am in the process of re-/moving the whole error handling logic. It's all over the place and can be handled better when creating providers the first time.